Update to .NET 10 build images

sbomer opened the pull request:
This updates the libc requirement to:
- glibc 2.27
- musl libc 1.2.3

These images have LLVM 19.

Contributes to https://github.com/dotnet/runtime/issues/109939

sbomer committed:
Update to .NET 10 build images

This updates the libc requirement to:
- glibc 2.27
- musl libc 1.2.3

These images have LLVM 19.

richlander commented on eng/pipelines/common/templates/pipeline-with-resources.yml:
```diff
@@ -27,44 +27,44 @@ extends:
 
       linux_musl_x64:
-        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net9.0-cross-amd64-alpine
+        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-amd64-alpine
```
I thought we were going to move to `-musl`. Wouldn't it make sense to block this PR on that change?

sbomer committed:
Use musl tags, update docs

jkoritzinsky commented:
Looks like there's some issues with the ARM32 images with zlib-ng, in particular around intrinsics.

Other than that, this looks good to me.

sbomer committed:
Fix arm32 build

The NEON_AVAILABLE and NEON_HAS_LD4 checks in zlib-ng attempt to pass
-mfpu=neon via CMAKE_REQUIRED_FLAGS, but our toolchain.cmake sets
-mfpu=vfpv4 via add_compile_options, which overrides the former.

Work around this by disabling neon on arm32 linux, matching .NET 9.

am11 commented on eng/pipelines/common/templates/pipeline-with-resources.yml:
```diff
@@ -109,17 +109,17 @@ extends:
 
       freebsd_x64:
-        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net9.0-cross-freebsd-13
+        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-freebsd-13-amd64
```
We have removed v13 images (i.e. existing images will stay in ACR, no new image will be produced)
```suggestion
        image: mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-freebsd-14-amd64
```

am11 commented on docs/workflow/using-docker.md:
```diff
@@ -26,26 +26,26 @@ The main Docker images are the most commonly used ones, and the ones you will pr
+| Azure Linux (x64) | Android Bionic             | x64           | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-android-amd64` |        *N/A*           |
+| Azure Linux (x64) | Android Bionic (w/OpenSSL) | x64           | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-android-openssl`     |        *N/A*           |
+| Azure Linux (x64) | Android Bionic (w/Docker)  | x64           | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-android-docker`      |        *N/A*           |
+| Azure Linux (x64) | FreeBSD 13                 | x64           | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-freebsd-13`    | `/crossrootfs/x64`     |
```
likewise
```suggestion
| Azure Linux (x64) | FreeBSD 14                 | x64           | `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net10.0-cross-freebsd-14`    | `/crossrootfs/x64`     |
```

sbomer committed:
Use FreeBSD 14 images

Co-authored-by: Adeel Mujahid <3840695+am11@users.noreply.github.com>

am11 committed:
Fix alpine arm32 build

am11 commented:
alpine arm32 build fixes: https://github.com/dotnet/runtime/compare/main...am11:runtime:feature/native/build-fixes

akoeplinger commented:
/ba-g test timeout is due to https://github.com/dotnet/runtime/issues/110285 which is fixed in main and closed so Build Analysis didn't link it.

ericstj commented:
@akoeplinger signed off and flipped to green, @sbomer is this ready to merge?

akoeplinger commented:
According to Outlook he's OOF until 12/19
